src/pages/studyView/addChartButton/geneLevelSelection/GeneLevelSelection.tsx

```diff
@@ -0,0 +1,199 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+import { Gene } from 'shared/api/generated/CBioPortalAPI';
+import { GenomicChart } from 'pages/studyView/StudyViewPageStore';
+import { observer } from 'mobx-react';
+import autobind from 'autobind-decorator';
+import { action, computed, observable } from 'mobx';
+import styles from './styles.module.scss';
+import ReactSelect from 'react-select';
+import { SingleGeneQuery } from 'shared/lib/oql/oql-parser';
+import { GeneReplacement } from 'shared/components/query/QueryStore';
+import OQLTextArea, {
+    GeneBoxType,
+} from 'shared/components/GeneSelectionBox/OQLTextArea';
+import classnames from 'classnames';
+import MobxPromise from 'mobxpromise';
+import { MakeMobxView } from 'shared/components/MobxView';
+import LoadingIndicator from 'shared/components/loadingIndicator/LoadingIndicator';
+import ErrorMessage from 'shared/components/ErrorMessage';
+
+export interface IGeneLevelSelectionProps {
+    molecularProfileOptionsPromise: MobxPromise<
+        {
+            value: string[];
+            label: string;
+            profileName: string;
+            description: string;
+        }[]
+    >;
+    submitButtonText: string;
+    onSubmit: (charts: GenomicChart[]) => void;
+}
+
+@observer
+export default class GeneLevelSelection extends React.Component<
+    IGeneLevelSelectionProps,
+    {}
+> {
+    @observable private geneInput: string = '';
+    @observable private validGenes: Gene[] = [];
+    @observable private isQueryInvalid: boolean = true;
+    @observable private hasOQL: boolean = false;
+    @observable private selectedProfileOption?: {
+        value: string[];
+        label: string;
+        profileName: string;
+        description: string;
+    };
+
+    public static defaultProps = {
+        disableGrouping: false,
+    };
+
+    @autobind
+    @action
+    private onAddChart() {
+        if (this.selectedOption !== undefined) {
+            const charts = this.validGenes.map(gene => {
+                return {
+                    name:
+                        gene.hugoGeneSymbol +
+                        ': ' +
+                        this.selectedOption!.profileName,
+                    description: this.selectedOption!.description,
```

Comment:
since you are checking for undefined above, i would think you don't need ! after this.selectedOption.  i think ts will know.

Reply:
Its throwing error if I remove it. May be because its in `map`